pkg/resource/deploy/step_generator.go

```diff
@@ -72,10 +72,34 @@ func (sg *stepGenerator) isTargetedUpdate() bool {
 	return sg.updateTargetsOpt != nil || sg.replaceTargetsOpt != nil
 }
 
+// isTargetedForUpdate checks if a urn is target for update.
+//
+// Note: to accommodate the `--target-dependents` flag,
+// `isTargetedForUpdateOrAddIfDep` should probably be called instead.
 func (sg *stepGenerator) isTargetedForUpdate(urn resource.URN) bool {
 	return sg.updateTargetsOpt == nil || sg.updateTargetsOpt[urn]
 }
 
+// isTargetedForUpdateOrAddIfDep checks if a target should be updated, updating
+// `sg.isTargetedForUpdateOrAddIfDep` as necessary for `--target-dependents` to be
+// accurate.
+func (sg *stepGenerator) isTargetedForUpdateOrAddIfDep(urn resource.URN, dependencies []resource.URN) bool {
```

Comment:
Nit: stylistically I would perhaps break it into two functions if it does have to mutate `sg.updateTargetsOpt`. It "looks" like a predicate because it returns bool, but it does mutate. I read the comment, then read the implementation, then had to read all the sites that touch `updateTargetsOpt` to figure out what it does. It seems like it does a sensible thing after all. 

Perhaps something like this?

```go
if sg.isAnyTargetedForUpdate(dependencies) {
   sg.targetForUpdate(urn) // more obviously mutating
}
sg.isTargetedForUpdate(urn)
```

Reply:
I spent way too long trying to come up with a name that made sense for this function. I was hoping it would read `(isTargetedForUpdate)Or(AddIfDep)`. The problem is that I have two separate ideas (`isTargetedForUpdate` and `addIfDep`) that I want to always be called together. I forced this by putting them in one function, but then there are naming problems. I can break these out, I'm just worried that someone (or me later) will forget to call one. Thanks for the feedback.

Reply:
Eh, don't sweat it it's a really small thing. Code will get rewritten... ultimately our defense is code reviews and thoughtful tests that remind when something is off.